diff -urN Arduino-1.8.10-orig/arduino-core/src/cc/arduino/utils/network/FileDownloaderCache.java Arduino-1.8.10/arduino-core/src/cc/arduino/utils/network/FileDownloaderCache.java
--- Arduino-1.8.10-orig/arduino-core/src/cc/arduino/utils/network/FileDownloaderCache.java	2019-09-13 13:21:47.000000000 +0300
+++ Arduino-1.8.10/arduino-core/src/cc/arduino/utils/network/FileDownloaderCache.java	2019-10-03 18:01:22.312672847 +0300
@@ -40,7 +40,6 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import processing.app.BaseNoGui;
 import processing.app.PreferencesData;
 import processing.app.helpers.FileUtils;
 
@@ -78,16 +77,19 @@
     }
     PreferencesData.set(CACHE_ENABLE_PREFERENCE_KEY, Boolean.toString(enableCache));
 
-    final File settingsFolder;
-    settingsFolder = BaseNoGui.getSettingsFolder();
-    if (settingsFolder != null) {
-      cacheFolder = Paths.get(settingsFolder.getPath(), "cache")
-        .toString();
+    final String xdgCacheHome;
+    final String cacheHome;
+
+    xdgCacheHome = System.getenv("XDG_CONFIG_HOME");
+
+    if(xdgCacheHome != null && !xdgCacheHome.isEmpty()) {
+      cacheHome = xdgCacheHome;
     } else {
-      enableCache = false;
-      cacheFolder = null;
-      log.error("The cache will disable because the setting folder is null, cannot generate the cache path");
+      cacheHome = Paths.get(System.getenv("HOME"), ".cache").toString();
     }
+
+    cacheFolder = Paths.get(cacheHome, "arduino").toString();
+
     final Path pathCacheInfo = getCachedInfoPath();
     log.info("Cache folder {}", cacheFolder);
     try {
diff -urN Arduino-1.8.10-orig/arduino-core/src/processing/app/Platform.java Arduino-1.8.10/arduino-core/src/processing/app/Platform.java
--- Arduino-1.8.10-orig/arduino-core/src/processing/app/Platform.java	2019-09-13 13:21:47.000000000 +0300
+++ Arduino-1.8.10/arduino-core/src/processing/app/Platform.java	2019-10-03 17:30:25.896339678 +0300
@@ -76,8 +76,25 @@
 
   public File getSettingsFolder() throws Exception {
     // otherwise make a .processing directory int the user's home dir
-    File home = new File(System.getProperty("user.home"));
-    File dataFolder = new File(home, ".arduino15");
+    File home = new File(System.getenv("HOME"));
+    File oldDataFolder = new File(home, ".arduino15");
+    File dataFolder;
+
+    if (oldDataFolder.isDirectory()) {
+      dataFolder = oldDataFolder;
+    } else {
+      File configHome;
+      String xdgConfigHome = System.getenv("XDG_CONFIG_HOME");
+
+      if(xdgConfigHome != null && !xdgConfigHome.isEmpty()) {
+        configHome = new File(xdgConfigHome);
+      } else {
+        configHome = new File(home, ".config");
+      }
+
+      dataFolder = new File(configHome, "arduino15");
+    }
+
     return dataFolder;
 
     /*
diff -urN Arduino-1.8.10-orig/build/shared/manpage.adoc Arduino-1.8.10/build/shared/manpage.adoc
--- Arduino-1.8.10-orig/build/shared/manpage.adoc	2019-09-13 13:21:47.000000000 +0300
+++ Arduino-1.8.10/build/shared/manpage.adoc	2019-10-03 17:34:57.941811082 +0300
@@ -219,7 +219,7 @@
 -----
 *%LOCALAPPDATA%/Arduino15/preferences.txt* (Windows)::
 *~/Library/Arduino15/preferences.txt* (Max OS X)::
-*~/.arduino15/preferences.txt* (Linux)::
+*~/.config/arduino15/preferences.txt* (Linux)::
 	This file stores the preferences used for the IDE, building and
 	uploading sketches.
 
