diff -urN miraclecast-9999.orig/CMakeLists.txt miraclecast-9999/CMakeLists.txt
--- miraclecast-9999.orig/CMakeLists.txt	2022-04-07 18:08:13.345355600 +0300
+++ miraclecast-9999/CMakeLists.txt	2022-04-07 18:09:35.255181627 +0300
@@ -12,6 +12,7 @@
 add_definitions(-D_GNU_SOURCE)
 
 SET(BUILD_BINDIR "${CMAKE_INSTALL_PREFIX}/bin")
+OPTION(ENABLE_SYSTEMD "Enable Systemd support" ON)
 OPTION(BUILD_ENABLE_DEBUG "Enable Debug" ON )
 OPTION(RELY_UDEV "Rely in udev tag to select device" OFF )
 OPTION(BUILD_TESTS "Enable TEST" ON )
@@ -28,7 +29,14 @@
 find_package(PkgConfig)
 pkg_check_modules (GLIB2 REQUIRED glib-2.0)
 pkg_check_modules (UDEV REQUIRED libudev)
-pkg_check_modules (SYSTEMD REQUIRED libsystemd)
+if(ENABLE_SYSTEMD)
+    pkg_check_modules (SYSTEMD REQUIRED libsystemd)
+    SET(SESSION_LIBRARIES "${SYSTEMD_LIBRARIES}")
+else(ENABLE_SYSTEMD)
+    pkg_check_modules (ELOGIND REQUIRED libelogind)
+    include_directories ("${ELOGIND_INCLUDEDIR}")
+    SET(SESSION_LIBRARIES "${ELOGIND_LIBRARIES}")
+endif(ENABLE_SYSTEMD)
 
 CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/config.h.cmake ${CMAKE_BINARY_DIR}/config.h)
 
diff -urN miraclecast-9999.orig/config.h.cmake miraclecast-9999/config.h.cmake
--- miraclecast-9999.orig/config.h.cmake	2022-04-07 18:08:13.346355638 +0300
+++ miraclecast-9999/config.h.cmake	2022-04-07 17:49:05.009130709 +0300
@@ -1,6 +1,7 @@
 #ifndef CONFIG_H
 #define CONFIG_H
 
+#cmakedefine ENABLE_SYSTEMD
 #cmakedefine BUILD_BINDIR "@BUILD_BINDIR@"
 #cmakedefine RELY_UDEV @RELY_UDEV@
 #cmakedefine IP_BINARY @IP_BINARY@
diff -urN miraclecast-9999.orig/src/shared/CMakeLists.txt miraclecast-9999/src/shared/CMakeLists.txt
--- miraclecast-9999.orig/src/shared/CMakeLists.txt	2022-04-07 18:08:13.349355751 +0300
+++ miraclecast-9999/src/shared/CMakeLists.txt	2022-04-07 18:10:09.595287158 +0300
@@ -1,7 +1,5 @@
 set(CMAKE_C_FLAGS "-std=gnu11") 
 
-find_package(PkgConfig)
-pkg_check_modules (SYSTEMD REQUIRED systemd>=213)
 set(miracle-shared_SOURCES rtsp.h
                              rtsp.c 
                              shl_dlist.h 
@@ -18,7 +16,7 @@
                              wpas.h 
                              wpas.c)
 add_library(miracle-shared STATIC ${miracle-shared_SOURCES})
-target_link_libraries (miracle-shared systemd)
+target_link_libraries (miracle-shared ${SESSION_LIBRARIES})
 
 
 ########### install files ###############
