--- a/cmake/opentrack-i18n.cmake	2023-02-01 12:54:20.891328304 +0200
+++ b/cmake/opentrack-i18n.cmake	2023-02-01 12:55:00.019718863 +0200
@@ -57,9 +57,6 @@
 endfunction()
 
 function(otr_merge_translations)
-    otr_escape_string(i18n-pfx "${opentrack-i18n}")
-    install(CODE "file(REMOVE_RECURSE \"\${CMAKE_INSTALL_PREFIX}/${i18n-pfx}\")")
-
     foreach(i ${opentrack_all-translations})
         get_property(ts-files GLOBAL PROPERTY "opentrack-ts-files-${i}")
         get_property(lrelease-binary TARGET "${Qt5_LRELEASE_EXECUTABLE}" PROPERTY IMPORTED_LOCATION)
