diff --git a/cmake/opentrack-install.cmake.orig b/cmake/opentrack-install.cmake
index 3983c82..5aac983 100644
--- a/cmake/opentrack-install.cmake.orig
+++ b/cmake/opentrack-install.cmake
@@ -23,8 +23,7 @@ endfunction()
 
 otr_install_dir("${opentrack-doc}" ${CMAKE_SOURCE_DIR}/3rdparty-notices)
 otr_install_dir("${opentrack-doc}" "${CMAKE_SOURCE_DIR}/settings" "${CMAKE_SOURCE_DIR}/contrib")
-
-INSTALL(CODE "FILE(MAKE_DIRECTORY \${CMAKE_INSTALL_PREFIX}/\${opentrack-hier-libexec}/presets)")
+otr_install_dir("${opentrack-libexec}" "${CMAKE_SOURCE_DIR}/presets")
 
 if(WIN32)
     otr_install_misc(. FILES "${CMAKE_SOURCE_DIR}/bin/qt.conf")
